pageserver/src/page_cache.rs

```diff
@@ -95,11 +91,15 @@ const MAX_USAGE_COUNT: u8 = 5;
 ///
 #[derive(Debug, PartialEq, Eq, Clone)]
 enum CacheKey {
-    MaterializedPage {
```

Comment:
Is is actually good idea to use the same cache both for ephemeral and layer files. May be maintain twoseparate caches, with the same implementation parameterized by key type?

Reply:
I think it is better to use the same cache for as many things as possible. That way, the memory can be dynamically allocated to where it's needed the most. Maybe we should have different weighting in the eviction algorithm and evict non-dirty pages more eagerly than dirty ephemeral pages, or something like that. But would need more performance testing and analysis to figure out what the optimum is.